### PR TITLE
Implemented basic functionality for Login and Signup Buttons and their buttons.

### PR DESCRIPTION
**Pull request by ChrisRPeterson**
- Clicking create an account button opens a modal with required text fields
- Clicking signup will create a user and a database document that stores that user's info including their UID, then the modal will close.
- Clicking cancel will close the modal.
- Login button will log a user in if the username and password are correct
- Logout button will logout the user
- Things that are not included in this PR
    - Error handling that renders anything to the screen
    - Styling
    - Basically, these aren't done yet, but just implement basic functionality
    - There is a non-crashing error when opening a modal. I think this is related to using React strict-mode with Material-UI modals. Not entirely sure what's up.

- Added userInfo to UserContext (this is what will hold userInfo you are more likely to use, such as bio), if there is no user it will be set to null, so be sure to add checks if you use it!

### CODE REVIEW

**Comment by ppattamasaevi** [comment]
Thanks for implementing the login/logout feature!
